### ISSUE TITLE
no .bin executable installed when installing with pnpm, node_modules/.bin doesn't contain bugsnag-cli

### ISSUE THREAD

**Issue by andylacko** [bugsnag/bugsnag-cli#108]
### Describe the bug
no .bin executable installed when installing with pnpm 

### Steps to reproduce
pnpm install @bugsnag/cli

### Environment
* `bugsnag-cli` version: 2.3.0
* Platform: macOS

**Comment by matthewjhowells** [comment]
Hi @, 

We’ve added an item to our backlog to support [pnpm](https://pnpm.io/). While we don’t currently have an ETA on when this could be looked at, we will update this issue once we have any updates on this. 

It’s worth noting that we do not officially support installing BugSnag packages via `pnpm` as none of our packages are tested against this package manager. In the meantime, we would recommend installing the `@bugsnag/cli` package using `npm` as we have verified that `npm` will install the bugsnag-cli executable in `.bin` as expected.

**Comment by cwhenderson20** [comment]
I'm seeing the same issue with yarn 3/4. Yarn [strictly expects](https://github.com/yarnpkg/berry/issues/6045#issuecomment-1871140280) packages to declare their binaries in the package.json metadata. I was following the automatic installation for React Native, which now uses Yarn 3 by default, but the scripts that get added to the root package.json fail with:

```
yarn bugsnag:create-build
command not found: bugsnag-cli
```

**Comment by mclack** [comment]
Hi @cwhenderson20 

Work on this feature request has started, which includes support for `pnpm` and `yarn`, and will available in the next `bugsnag-cli` release.

Although we do not currently have an ETA for the release, we will make sure to update this thread as soon as it is available.

**Comment by rossipedia** [comment]
Just recently found this thread trying to get `yarn dlx -p @bugsnag/cli bugsnag-cli` working and banging my head against it for a while 😅 

Having support for alternative package managers (`yarn` / `pnpm`) would be much appreciated 🙏

**Comment by mclack** [comment]
Hi 👋🏻

We have now released v2.4.0 of the `bugsnag-cli`, which should ensure the binary is correctly installed when using `pnpm` and `yarn`: https://github.com/bugsnag/bugsnag-cli/releases/tag/v2.4.0

As such, we are now going to close this issue. Please reach out again if you continue to have issues with this, and feel free to open a ticket with us directly by contacting support@bugsnag.com.

**Comment by attenzione** [comment]
Hi @mclack,

I'm currently migrating from [`webpack-bugsnag-plugins`](https://github.com/bugsnag/bugsnag-source-maps/issues/97) to `bugsnag-cli` and encountered an issue while executing the command.

You're declaring the bin as:
```json
"bin": {
  "bugsnag-cli": "bin/bugsnag-cli"
}
```
However, the bin file is not placed inside the package at `node_modules/@bugsnag/cli`, where `yarn` expects it. Instead, the bin file is downloaded and placed directly in `node_modules/.bin`, but this action should be performed by `yarn` or `npm`, and as a result a symlink is created.

Currently, in my `node_modules/.bin`, I have:
```bash
lrwxrwxrwx   1       32 Dec 19  2023 autoprefixer -> ../autoprefixer/bin/autoprefixer
lrwxrwxrwx   1       42 Sep 19 09:46 browsers -> ../@puppeteer/browsers/lib/cjs/main-cli.js
lrwxrwxrwx   1       22 Dec 19  2023 browserslist -> ../browserslist/cli.js
-rwxr-xr-x   1 10186904 Sep 26 12:38 bugsnag-cli
lrwxrwxrwx   1       20 Dec 19  2023 cssesc -> ../cssesc/bin/cssesc
lrwxrwxrwx   1       22 Dec 19  2023 envinfo -> ../envinfo/dist/cli.js
...
```
As you can see, `bugsnag-cli` is a physical file, not a symlink (it’s the only one like this — all others are symlinks).

Yes, executing `/node_modules/.bin/bugsnag-cli` works, but this is not the correct approach. 
The correct way should be using:
```
yarn run bugsnag-cli
```
or simply:
```
yarn bugsnag-cli
```
Both of these result in the following error:
```
Error: Cannot find module '/node_modules/@bugsnag/cli/bin/bugsnag-cli'
```

**Comment by hannah-smartbear** [comment]
Hi @attenzione,

Testing this on my side, it is indeed the executable that gets installed to `node_modules/.bin` rather than a symlink, but `yarn` still seems to be able to find the executable - I’m not able to reproduce the issue you are seeing when running `yarn bugsnag-cli` . Please can you confirm which version of `yarn` you are using?

**Comment by hannah-smartbear** [comment]
Hi @attenzione,

Thanks for providing your Yarn version. We have reproduced this, and we are going to investigate implementing a fix to ensure Yarn is supported. We will be sure to let you know of any updates.

**Comment by attenzione** [comment]
Hi @joshedney,
Thanks for the update on the issue.

I’ve tried the new version and successfully got the working command:

```bash
git:(feature/bugsnag-cli) yarn bugsnag-cli --version
2.6.4-alpha

git:(feature/bugsnag-cli) yarn bugsnag-cli
Usage: bugsnag-cli <command>
...
```

Also, as you're working on this part of the package, I recommend dropping the `axios` dependency. Since **Node v18**, you can use the native `fetch` api.

The code below from [Medium](https://medium.com/deno-the-complete-reference/download-file-with-fetch-in-node-js-57dd370c973a), demonstrates how you can achieve this:

```js
import { createWriteStream } from "fs";
import { Readable } from "stream";

const url = "https://cdn.freebiesupply.com/logos/large/2x/nodejs-1-logo-png-transparent.png";
const fileName = url.split("/").pop();
const resp = await fetch(url);

if (resp.ok && resp.body) {
  let writer = createWriteStream(fileName);
  Readable.fromWeb(resp.body).pipe(writer);
}
```

Would you like me to create a new issue for this?

**Comment by mclack** [comment]
Hi @attenzione 

Thanks for confirming that.

We have now released v2.6.2 of `bugsnag-cli`, which ensures support for `yarn`, and also replaces the `axios` dependency with the native `fetch` API: https://github.com/bugsnag/bugsnag-cli/releases/tag/v2.6.2

Please let us know if you encounter any further issues with this!
